src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.Workable;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private MachineProducer machineProducer;
+
+    @Override
+    public List getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            machineProducer = new BulldozerProducer();
+            return machineProducer.get();
+        } else if (type == Truck.class) {
+            machineProducer = new TruckProducer();
+            return machineProducer.get();
+        } else if (type == Excavator.class) {
+            machineProducer = new ExcavatorProducer();
```

Comment:
краще в класі оголосити три змінні. new ....Producer()
і тут можна буде так віддавати ліст 
```suggestion
            return new ArrayList(excavatorProducer.get());
```
так ми не будемо створювати об'єктів класів продюсерів при кожному виклику заново